pkg/apis/mesh/types.go

```diff
@@ -81,43 +87,118 @@ const (
-	string(MeshConfigLoadBalancingMethodLeastConn):                  {},
-	string(MeshConfigLoadBalancingMethodLeastTime):                  {},
-	string(MeshConfigLoadBalancingMethodLeastTimeLastByte):          {},
-	string(MeshConfigLoadBalancingMethodLeastTimeLastByteInflight):  {},
-	string(MeshConfigLoadBalancingMethodRandom):                     {},
-	string(MeshConfigLoadBalancingMethodRandomTwo):                  {},
-	string(MeshConfigLoadBalancingMethodRandomTwoLeastConn):         {},
-	string(MeshConfigLoadBalancingMethodRandomTwoLeastTime):         {},
-	string(MeshConfigLoadBalancingMethodRandomTwoLeastTimeLastByte): {},
+// Access Control Modes.
+const (
+	AccessControlModeDeny  = "deny"
+	AccessControlModeAllow = "allow"
+)
+
+// AccessControlModes are the supported access control modes.
+var AccessControlModes = map[string]struct{}{
+	AccessControlModeAllow: {},
+	AccessControlModeDeny:  {},
 }
 
+// NGINX error log levels.
+const (
+	NginxErrorLogLevelDebug  = "debug"
+	NginxErrorLogLevelInfo   = "info"
+	NginxErrorLogLevelNotice = "notice"
+	NginxErrorLogLevelWarn   = "warn"
+	NginxErrorLogLevelError  = "error"
+	NginxErrorLogLevelCrit   = "crit"
+	NginxErrorLogLevelAlert  = "alert"
+	NginxErrorLogLevelEmerg  = "emerg"
+)
+
+// NGINX log formats.
+const (
+	NginxLogFormatDefault = "default"
+	NginxLogFormatJSON    = "json"
+)
+
 // NGINXLogFormats are the supported NGINX log formats.
 var NGINXLogFormats = map[string]struct{}{
-	string(MeshConfigNginxLogFormatDefault): {},
-	string(MeshConfigNginxLogFormatJson):    {},
+	NginxLogFormatDefault: {},
+	NginxLogFormatJSON:    {},
 }
 
+// NGINX load balancing methods.
+const (
+	RoundRobin                 = "round_robin"
+	LeastConn                  = "least_conn"
+	LeastTime                  = "least_time"
+	LeastTimeLastByte          = "least_time last_byte"
+	LeastTimeLastByteInflight  = "least_time last_byte inflight"
+	Random                     = "random"
+	RandomTwo                  = "random two"
+	RandomTwoLeastConn         = "random two least_conn"
+	RandomTwoLeastTime         = "random two least_time"
+	RandomTwoLeastTimeLastByte = "random two least_time=last_byte"
+)
+
+// LoadBalancingMethods are the available NGINX load balancing methods.
+var LoadBalancingMethods = map[string]struct{}{
+	RoundRobin:                 {},
+	LeastConn:                  {},
+	LeastTime:                  {},
+	LeastTimeLastByte:          {},
+	LeastTimeLastByteInflight:  {},
+	Random:                     {},
+	RandomTwo:                  {},
+	RandomTwoLeastConn:         {},
+	RandomTwoLeastTime:         {},
+	RandomTwoLeastTimeLastByte: {},
+}
+
+// Kubernetes environments.
+const (
+	Kubernetes = "kubernetes"
+	Openshift  = "openshift"
+)
+
 // Environments are the supported kubernetes environments.
 var Environments = map[string]struct{}{
-	string(Kubernetes): {},
-	string(Openshift):  {},
+	Kubernetes: {},
+	Openshift:  {},
 }
```

Comment:
Couldn't this be a slice of `{Kubernetes, Openshift}` instead of a map of structs?

Reply:
These are all maps for easy lookup using keys.